openml/extensions/sklearn/extension.py

```diff
@@ -1346,7 +1346,7 @@ def _can_measure_cputime(self, model: Any) -> bool:
         # check the parameters for n_jobs
         n_jobs_vals = SklearnExtension._get_parameter_values_recursive(model.get_params(), "n_jobs")
         for val in n_jobs_vals:
-            if val is not None and val != 1:
+            if val is not None and val != 1 and val != "deprecated":
```

Comment:
Could you briefly explain why this is necessary?

Reply:
This ties with the next comment you made.

The common issue arises from the `deprecated` value assigned to the `n_jobs` parameter for `KMeans` for sklearn 0.23 and above.

The difference is that in this [unit test](https://github.com/openml/openml-python/blob/dcf2d22e4b92b6afc5bfb15b69c4f8d37f15d57a/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L167) the `n_jobs_val` is verified explicitly through `assertEqual` as part of the `fixture\_parameters`. Whereas in this [unit test](https://github.com/openml/openml-python/blob/dcf2d22e4b92b6afc5bfb15b69c4f8d37f15d57a/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L1779) the check is happening through a function defined under `sklearn/extension.py` where during `_run_model_on_fold` the cpu time estimates are I believe calculated/included, conditioned on the boolean returned by the function `_can_measure_cputime`, i.e., the function you highlighted in this review comment.

Hence, I included `deprecated` to be handled at this level in a generic way.

Reply:
Okay, I think I got how this works. I'm fine with it. However, we need to rethink the design here I guess because if a job uses OpenMP we can no longer measure CPU time :(